test/parallel/test-assert.js

```diff
@@ -447,6 +447,7 @@ assert.throws(makeBlock(thrower, TypeError));
   } catch (e) {
     threw = true;
     assert.ok(e instanceof a.AssertionError);
+    assert.ok(!e.stack.includes('at Function.doesNotThrow'));
```

Comment:
As requested, I removed the unnecessary test, and enriched an existing one.

Without stack trace cleaning:
```
AssertionError [ERR_ASSERTION]: Got unwanted exception.
[object Object]
    at new AssertionError (internal/errors.js:320:11)
    at innerFail (assert.js:74:9)
    at expectsNoError (assert.js:464:5)
    at Function.doesNotThrow (assert.js:488:3)
    at Object.<anonymous> (D:\Workspaces\perso\node\test\parallel\test-assert.js:450:7)
```

With cleaning:
```
AssertionError [ERR_ASSERTION]: Got unwanted exception.
[object Object]
    at Object.<anonymous> (D:\Workspaces\perso\node\test\parallel\test-assert.js:446:7)
```